### ISSUE TITLE
Add Support for Saving and Reusing Filter Settings as Profiles

### ISSUE THREAD

**Issue by quamt** [rigaya/Aviutl-clcufilters#1]
I suggest an enhancement to add a feature that allows users to save their filter settings as reusable profiles. This functionality would enable users to quickly select commonly used settings without manually adjusting them each time, making the workflow more efficient and user-friendly.

Proposed Solution:

Introduce a "Save Profile" option within the filter settings, allowing users to save their current settings under a custom name.
Add a "Load Profile" option that lets users select from previously saved profiles and apply those settings instantly.
Provide a "Manage Profiles" feature that allows users to delete or rename saved profiles as needed.
Benefits:

Increased Efficiency: Users often apply the same filter configurations across different projects. By saving settings as profiles, they can use these configurations with a single click, reducing repetitive adjustments and enhancing workflow speed.

Consistency Across Projects: Saved profiles ensure consistency and accuracy for users who work on multiple projects requiring similar settings. This minimizes the risk of errors from manual input.

Improved Usability for New Users: New users can benefit from predefined or community-shared profiles (if provided). This makes experimenting with different settings and understanding their impact easier, fostering faster learning and adoption.

Customization and Flexibility: Users can tailor profiles to fit different needs, from quick adjustments to in-depth edits, adding a layer of flexibility without complicating the interface.

This feature would be a valuable addition to Aviutl-clcufilters, further enhancing its usability and appeal to both new and experienced users.

Thank you for considering this suggestion. I appreciate your efforts to improve the project continually.

**Comment by rigaya** [comment]
Thank you for your proposal.

I also think it can be useful as now we have too many parameters.

I think I'll first try to add it in more simple way by saving and loading profile to and from a file by "Save Profile" and "Load Profile" button, but skipping to add "Manage Profiles", because it can be simply done by deleting and renaming the profile files from the Windows explorer directly.

**Comment by quamt** [comment]
Thank you for considering my suggestion and for your prompt response.

I appreciate your plan to implement a simplified version of the profile-saving feature. I understand that managing profiles can be handled manually through Windows Explorer, and I'm excited to see how the initial version of this feature evolves.

Thank you again for your hard work.

**Comment by quamt** [comment]
I thought about it, but I don't knowf it's feasible.

However, I wanted to propose an additional enhancement to further improve the user experience: organizing the filter settings into categories or tabs. This approach could make it easier to navigate specific types of filters by groupingether, helping users quickly finey need.

_**Proposed Solution:**_

**Tabs or Sections:** Introduce tabs or collapsible sections within the interface, each dedicated to a specific filter category (e.g., "Resize," "Deinterlace," "Noise Removal", etc.).

**Quick Access:** Users would be able to access relevant filters facanolling through a long list.

_**Benefits:**_

**Improved Usability:** Categorizing filters would simplify navigation, making it easier for users to find the exact tools they need.

**Reduced Clutter:** Organizing filtersnecessary toolsvisual clutter, making the interface more approachable, especially for new users.

**Enhanced Workflow:** It allows users to focus on specific tasks more efficiently, as they can quickly switch to the appropriate filter category.

I understand this might be more complex to implement, and it could require significant changes to the UI. I just wanted to share the idea in case it aligns with your developmeam sharingknow that encoders have priority! :-)

_Thank you again for considering this additional suggestion!_

**Comment by rigaya** [comment]
Yes, I can understand the benefit of tabs, and I actually really want tab controls for this, as the number of filters and parameters are growing, thus the window is taking too much space.

However, I must say that I'm unsure if I'll try to add tabs (or sections) in the future.

The plugin window is created by Aviutl (plugin developer actually only set the window configs). Aviutl does not have option for tab control, and only has settings for trackbars and checkboxs. Current clcufilters uses some legacy win32api to change the window from the default to re-organize to current window and add extra comboboxs and buttons, but I'm not really familiar to win32api, so I'm not quite sure I can add tab control there. I know that it will take much time so I'm hesitating to try on it.

**Comment by quamt** [comment]
Thank you for providing insights into the technical limitations. 

I completely understand the challenges, especially with Aviutl’s restricted plugin interface and the complexities of using Win32API to modify the window beyond its native configuration.

Given these limitations, I wonder if implementing separate windows for each category of filters, rather than tabs within a single window, could serve as a possible workaround. While this approach might not be as streamlined, it could help organize the filters without requiring tab controls.

**Comment by rigaya** [comment]
Unfortunately, separating window is also difficult to handle the window messages, same as tab control.

Actually, next version is planned to add filter categories in labels.

Also, next version will open individual parameters of the filter only when the filter is in use (checked).

This aims 2 points below, assuming that there is little chance that many filters are in use at the same time.
- display fewer things at a time, to improve user visibility
- avoid window height continuing to get larger, which was reaching 1000pix, which will be difficult to use for users using 1080p or smaller display

![image](https://github.com/user-attachments/assets/1ec81e5d-84f2-4e02-9b40-f3a4448c4e3f)
